### PR TITLE
[v2] feat: sign-client events tests

### PR DESCRIPTION
**Pull request by ganchoradkov**
Tests added for the following events

- session_proposal
- session_update
- session_ping
- session_event
- session_delete

ref https://github.com/WalletConnect/walletconnect-monorepo/issues/1153

### CODE REVIEW

File: packages/sign-client/test/events.spec.ts [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+          
+          clients.A.disconnect({ topic: sessionA.topic, reason: getSdkError('USER_DISCONNECTED')});
+        } catch (e) {
+          reject(e);
+        }
+      })
+      deleteClients(clients);
+    });
+  });
+});
```
Comment (xzilja) [inline packages/sign-client/test/events.spec.ts]: Add new line here. This should be managed by prettier though, are you using eslint and prettier extensions in your ide?

File: packages/sign-client/test/events.spec.ts [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+      expect(pairingA.topic).to.eql(uriParams.topic);
+      expect(pairingA.relay).to.eql(uriParams.relay);
+
+      if (!pairingA) throw new Error("expect pairing A to be defined");
+
+      let sessionA: SessionTypes.Struct;
+      let sessionB: SessionTypes.Struct;
+
+      await Promise.all([
+        new Promise<void>((resolve, reject) => {
```
Comment (xzilja) [inline packages/sign-client/test/events.spec.ts:56]: Honestly, you probably don't need to specify return type `<void>` in tests

**Review by xzilja — APPROVED** [review]
Niiiice, left few minor suggestions

File: packages/sign-client/test/events.spec.ts [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+          } catch (error) {
+            reject(error);
+          }
+        })
+      ]);
+
+      deleteClients(clients)
+    });
+
+    it("session_update", async () => {
```
Comment (bkrem) [inline packages/sign-client/test/events.spec.ts]: Can we turn each event type into a `describe`, i.e. `describe("session_update", ...)` and then have this test inside with a description e.g. `it("emits and handles a valid session_update")`?

That way its more descriptive what the expectations are for this specific test, and lets us add regression tests or expected failure cases for each event type later.

**Review by bkrem — APPROVED** [review]

File: packages/sign-client/test/events.spec.ts (hunk 2 above)
Comment (ganchoradkov) [reply packages/sign-client/test/events.spec.ts:56]: It's return `<void>` otherwise `resolve()` expects a value

**Pull request merged**